### PR TITLE
New namespace package layout

### PR DESCRIPTION
**Pull request by spookylukey**
First stab at #79

### CODE REVIEW

**Comment by spookylukey** [comment]
Notes - see:

* https://packaging.python.org/guides/packaging-namespace-packages/#pkgutil-style-namespace-packages
* https://github.com/pypa/sample-namespace-packages/tree/master/pkgutil#practical-usage

**Comment by spookylukey** [comment]
Hmm, that's weird. I got exactly those failures first time I tried, turned out to be due to git 'fixing' up the whitespace in some of the files, since I have this in my global git config:

    [core]
      whitespace=fix,-indent-with-non-tab,trailing-space,cr-at-eol

So, I then overrode that locally:

    git config core.whitespace ''

and re did the 'git mv' commands. Now everything works locally, the tests pass. But not on Travis. So it seems that when I pushed, something else applied that same git whitespace config...

**Comment by spookylukey** [comment]
Hm, it seems that git is still actually rewriting the file when I do 'git mv', but on disk it just moves the existing file, so that what you have in the repo is different from what you have in your working directory. And it doesn't tell you that when you do git status. Nice...

File: fluent-syntax/setup.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from setuptools import setup
 
-setup(name='fluent',
+setup(name='fluent_syntax',
```
Comment (stasm) [inline fluent-syntax/setup.py]: I'd prefer to use `_` here (or `-`; practically they're the same), but unfortunately `fluent-bundle` is [already taken on PyPI](https://pypi.org/project/fluent-bundle/). I see three paths forward:

1. We go with `fluent_syntax` and find a different name for `fluent_bundle` in the same schema, e.g. `fluent_runtime`.
2. We go with `fluent_syntax` and re-use the existing `fluent` package name for `fluent_bundle`.
3. We use a different separator. According to [PEP 426](https://www.python.org/dev/peps/pep-0426/#name), periods are allowed, too. How about `fluent.syntax` here and `fluent.bundle` in #81?

Out of the three, I think I like the option no. 3 the most, but I'm open to other opinions.

Comment (spookylukey) [reply fluent-syntax/setup.py]: Option 3 also looks good to me. My choice of `_` was due to following the guidelines here - https://github.com/pypa/sample-namespace-packages/tree/master/pkgutil#practical-usage - but it sounds like a period could work too. We can try that - I'm concerned the combination with namespace packages might confuse something - and if there is a problem we can think again.

**Pull request merged**